src/timesync/timesyncd-bus.c

```diff
@@ -43,6 +45,58 @@ static int property_get_servers(
         return sd_bus_message_close_container(reply);
 }
 
+static int method_set_runtime_servers(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_strv_free_ char **valid_names = NULL;
+        _cleanup_strv_free_ char **msg_names = NULL;
+        Manager *m = userdata;
+        int r;
+
+        assert(m);
+        assert(message);
+
+        r = sd_bus_message_read_strv(message, &msg_names);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(name, msg_names) {
+                r = dns_name_is_valid_or_address(*name);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to check validity of NTP server name or address '%s': %m", *name);
+
+                if (r == 0) {
+                        log_error("Invalid NTP server name or address, ignoring: %s", *name);
+                        continue;
+                }
+
+                r = strv_extend(&valid_names, *name);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add runtime server '%s': %m", *name);
```

Comment:
the only error strv_extend() returns is ENOMEM, hence maybe just use `return log_oom()` here.

Reply:
fair enough!